addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -1301,6 +1301,11 @@ export class PosStore extends Reactive {
         this.ticket_screen_mobile_pane =
             this.ticket_screen_mobile_pane === "left" ? "right" : "left";
     }
+    getTableFromElement(el) {
+        return this.models["restaurant.table"].get(
+            [...el.classList].find((c) => c.includes("tableId")).split("-")[1]
+        );
+    }
```

Comment:
Since its a refactoring PR,  table can be retrieved easily by using attribute instead of classes.

`t-att-tableid="table.id`

And get it with:

`el.attributes.tableid.value`

```
getTableFromElement(el) {
    return this.pos.models["restaurant.table"].get(el.attributes.tableid.value);
}
```